src/Microsoft.Performance.SDK.Runtime/ColumnBuilding/Builders/EmptyColumnBuilder.cs

```diff
@@ -73,7 +78,13 @@ public override ToggleableColumnBuilder WithHierarchicalToggle<T>(
         return new ToggledColumnBuilder(
             [
                 new ToggledColumnBuilder.AddedToggle(toggleDescriptor,
-                    new HierarchicalDataColumn<T>(baseColumn.Configuration, projection, collectionProvider)),
+                    new HierarchicalDataColumn<T>(
+                        new ColumnConfiguration(this.baseColumn.Configuration)
+                        {
+                            Metadata = new ColumnMetadata(this.baseColumn.Configuration.Metadata) { Name = toggleDescriptor.Properties.ColumnName ?? this.baseColumn.Configuration.Metadata.Name},
```

Comment:
Can use the `with` keyword here if `ColumnMetadata` is a record

Reply:
yep, I am aware but didn't want to convert ColumnMetadata